src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1485,24 +1485,24 @@ public static bool IsAtEndOfPattern(this SyntaxTree syntaxTree, SyntaxToken left
             static bool IsAtEndOfSwitchStatementPattern(SyntaxToken leftToken)
             {
                 var node = leftToken.Parent;
+                if (node == null)
+                    return false;
 
                 // Walking up the tree for expressions such as 'case (((N.C.P $$'
-                while (node.IsParentKind(SyntaxKind.SimpleMemberAccessExpression))
-                {
-                    node = node.Parent;
-                }
+                if (node.IsParentKind(SyntaxKind.SimpleMemberAccessExpression))
+                    node = node.GetRequiredParent();
```

Comment:
while loop was bogus.  this would walk us up a member access expr we were in the middle of, not at teh end of.  if you're at the end, you only need to walk up one level.

Reply:
Sorry, I think I did the initial implementation of this. 😆 thanks for the fix!